.github/workflows/scan.yml

```diff
@@ -0,0 +1,31 @@
+name: GitHub Actions Security Analysis with zizmor 🌈
+
+on:
+  schedule:
+    # Daily
+    - cron: "23 7 * * *"
+  push:
+    paths:
+      - .github/workflows/*
+  pull_request:
+    paths:
+      - .github/workflows/*
+
+permissions: {}
+
+jobs:
+  zizmor:
+    name: Run zizmor 🌈
+    runs-on: ubuntu-latest
+    permissions:
+      security-events: write
+      # contents: read # only needed for private repos
+      # actions: read # only needed for private repos
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
```

Comment:
Is there an advantage to using the commit hash instead of `v5` like we do for other workflows?

e.g. https://github.com/ansible/ansible-documentation/blob/60c7262c1d645a4c84f92431c836c7a142935850/.github/workflows/reusable-build-docs.yaml#L43

Would this have any effect on dependabot?

Reply:
@felixfontein

> will even update the comment so that version and hash match. (That sometimes doesn't work for reasons I don't know, so reviewing a Dependabot PR gets a bit harder.)

I think this requires having the tag and the commit hash match before Dependabot runs. Otherwise, it'll probably assume its some unrelated comment and keep it as is.

> The main downside is that there will be a lot more Dependabot PRs, since every patch release of an action will require an update.

Though, it can be configured to bundle updates and run less often.

Alternatively, I'd suggest using https://gha-update.rtfd.io for a more fine-grained control, instead of Dependabot.

I've also seen some projects starting to use `pinact`: https://github.com/marketplace/actions/pinact.